This internal patch addresses an apparent Juno regression related to
Launchpad bug 1252403.  It's not suitable for the upstream but a bug
report will be filed to cover the regression and eventually this patch
should be replaced with something official.

--- horizon-2015.1.2/openstack_dashboard/api/neutron.py.orig	2015-12-12 16:12:01.520290169 -0500
+++ horizon-2015.1.2/openstack_dashboard/api/neutron.py	2015-12-12 16:14:28.179524164 -0500
@@ -431,8 +431,7 @@
                           in ext_net_ids)]
         reachable_subnets = set([p.fixed_ips[0]['subnet_id'] for p in ports
                                 if ((p.device_owner in
-                                     ROUTER_INTERFACE_OWNERS)
-                                    and (p.device_id in gw_routers))])
+                                     ROUTER_INTERFACE_OWNERS))])
         # we have to include any shared subnets as well because we may not
         # have permission to see the router interface to infer connectivity
         shared = set([s.id for n in network_list(self.request, shared=True)
